File: src/main/java/xyz/lncvrt/galaxyboxpvp/commands/Autocompress.java
package xyz.lncvrt.galaxyboxpvp.commands;

import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvP;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

import java.util.UUID;

import static org.bukkit.ChatColor.GREEN;
import static org.bukkit.ChatColor.RED;

public record Autocompress(GalaxyBoxPvP plugin) implements CommandExecutor {

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, String[] args) {
        if (!(sender instanceof Player player)) {
            sender.sendMessage(RED + "This command can only be executed by a player.");
            return true;
        }

        UUID playerId = player.getUniqueId();

        boolean currentStatus = plugin.autoCompressStatus.getOrDefault(playerId, false);
        plugin.autoCompressStatus.put(playerId, !currentStatus);

        if (!currentStatus) {
            player.sendMessage(GREEN + "Auto-compress enabled.");
        } else {
            player.sendMessage(RED + "Auto-compress disabled.");
        }
        return true;
    }
}

File: src/main/kotlin/xyz/lncvrt/galaxyboxpvp/commands/Autocompress.kt
package xyz.lncvrt.galaxyboxpvp.commands

import org.incendo.cloud.kotlin.MutableCommandBuilder
import org.sayandev.stickynote.bukkit.command.BukkitCommand
import org.sayandev.stickynote.bukkit.command.BukkitSender
import org.sayandev.stickynote.bukkit.extension.sendComponent
import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvPPlugin

object Autocompress : BukkitCommand("autocompress") {
    override fun rootBuilder(builder: MutableCommandBuilder<BukkitSender>) {
        builder.handler { context ->
            val player = context.sender().player() ?: return@handler
            val playerId = player.uniqueId
            val instance = GalaxyBoxPvPPlugin.getInstance()

            val currentStatus = instance.autoCompressStatus.getOrDefault(playerId, false) ?: return@handler
            instance.autoCompressStatus.put(playerId, !currentStatus)

            if (!currentStatus) {
                player.sendComponent("<green>Auto-compress enabled.")
            } else {
                player.sendComponent("<red>Auto-compress disabled.")
            }
        }
    }
}
